### ISSUE TITLE
Elasticsearch cluster mode

### ISSUE THREAD

**Issue by freignat91** [appcelerator-archive/amp#1384]
I got issue creating elasticsearch service with more than one replicas:

```
ID                  NAME                      IMAGE                                  NODE                DESIRED STATE       CURRENT STATE           ERROR                       PORTS
pjvpu2i2asjh        amp_elasticsearch.1       appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Ready               Ready 1 second ago                                  
5uff674etla2         \_ amp_elasticsearch.1   appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Shutdown            Failed 2 seconds ago    "task: non-zero exit (1)"   
ljl2auy5kanr         \_ amp_elasticsearch.1   appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Shutdown            Failed 12 seconds ago   "task: non-zero exit (1)"   
0n12xb9injkh         \_ amp_elasticsearch.1   appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Shutdown            Failed 23 seconds ago   "task: non-zero exit (1)"   
zkrhtloag9wh         \_ amp_elasticsearch.1   appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Shutdown            Failed 33 seconds ago   "task: non-zero exit (1)"   
mywjdwvbuhw6        amp_elasticsearch.2       appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Ready               Ready 1 second ago                                  
qdr232tfm3yu         \_ amp_elasticsearch.2   appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Shutdown            Failed 2 seconds ago    "task: non-zero exit (1)"   
sgupg5d01c5w         \_ amp_elasticsearch.2   appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Shutdown            Failed 12 seconds ago   "task: non-zero exit (1)"   
kfd5nws9vmpk         \_ amp_elasticsearch.2   appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Shutdown            Failed 23 seconds ago   "task: non-zero exit (1)"   
t1ob0aj0bve7         \_ amp_elasticsearch.2   appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Shutdown            Failed 33 seconds ago   "task: non-zero exit (1)"   
tv6a0ge0ln73        amp_elasticsearch.3       appcelerator/elasticsearch-amp:5.4.1   fr-amp-manager1     Running             Running 3 minutes ago  
```

It works well with replicas 1

**Comment by ndegory** [comment]
logs would be helpful.
did you use the ampmon-cluster.1.stack.yml file?

**Comment by freignat91** [comment]
here the logs:
```
INFO - system memory is 8173M
INFO - set java heap size to 10%
Java Heap Size: 817 MB
memory lock limit:
64
64
Hard limit max open fd:1048576
Max open fd:1048576
resolving the container IP with Docker DNS...
this node ip is 10.0.0.7
waiting for other tasks to be available...
2 tasks found for unicast zen discovery (10.0.0.44,10.0.0.45)
INFO - setting user perms on elasticsearch data dir
INFO - running elasticsearch as user elastico
INFO - running gosu elastico elasticsearch
[2017-06-15T14:40:44,477][INFO ][o.e.n.Node               ] [] initializing ...
[2017-06-15T14:40:44,509][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [] uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupException: java.lang.IllegalStateException: failed to obtain node locks, tried [[/opt/elasticsearch/data/elasticsearch]] with lock id [0]; maybe these locations are not writable or multiple nodes were started without increasing [node.max_local_storage_nodes] (was [1])?
        at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:127) ~[elasticsearch-5.4.1.jar:5.4.1]
        at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:114) ~[elasticsearch-5.4.1.jar:5.4.1]
        at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:67) ~[elasticsearch-5.4.1.jar:5.4.1]
        at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:122) ~[elasticsearch-5.4.1.jar:5.4.1]
        at org.elasticsearch.cli.Command.main(Command.java:88) ~[elasticsearch-5.4.1.jar:5.4.1]

```

**Comment by freignat91** [comment]
ip link show: n(on node having failing ES container)
```
1792: vethc9fd308@if1791: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether c2:ee:24:43:7f:78 brd ff:ff:ff:ff:ff:ff link-netnsid 18
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether 02:2b:7a:06:05:66 brd ff:ff:ff:ff:ff:ff
1796: veth871cda6@if1795: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 86:2d:f9:04:36:11 brd ff:ff:ff:ff:ff:ff link-netnsid 19
4: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN mode DEFAULT group default 
    link/ether 02:42:62:4b:8d:f4 brd ff:ff:ff:ff:ff:ff
9: docker_gwbridge: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default 
    link/ether 02:42:64:3c:ab:fa brd ff:ff:ff:ff:ff:ff
1802: veth34e69ef@if1801: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 46:1d:89:be:51:36 brd ff:ff:ff:ff:ff:ff link-netnsid 22
11: veth1db3690@if10: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 96:79:83:5a:5f:bb brd ff:ff:ff:ff:ff:ff link-netnsid 1
1808: veth688d00e@if1807: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 2e:14:9e:96:15:7a brd ff:ff:ff:ff:ff:ff link-netnsid 23
1814: veth4c8ba00@if1813: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 96:1d:6e:00:f7:cf brd ff:ff:ff:ff:ff:ff link-netnsid 24
1818: veth54e6449@if1817: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether ea:65:b1:b1:9a:08 brd ff:ff:ff:ff:ff:ff link-netnsid 25
1822: veth9ab9c07@if1821: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 26:30:f2:e9:5b:8b brd ff:ff:ff:ff:ff:ff link-netnsid 26
1826: veth23044e0@if1825: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether a2:4e:ba:bb:b0:8e brd ff:ff:ff:ff:ff:ff link-netnsid 27
1832: veth6842d78@if1831: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether e6:54:c2:dc:f4:45 brd ff:ff:ff:ff:ff:ff link-netnsid 28
1836: vetheba2611@if1835: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 5e:c1:17:53:7b:b9 brd ff:ff:ff:ff:ff:ff link-netnsid 29
1840: vetheb26c6c@if1839: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether a2:a6:71:2a:6f:0d brd ff:ff:ff:ff:ff:ff link-netnsid 30
1850: veth8c61c51@if1849: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 0e:e1:44:ba:36:c8 brd ff:ff:ff:ff:ff:ff link-netnsid 12
2162: veth6d373b8@if2161: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 6a:bc:c9:92:c6:8d brd ff:ff:ff:ff:ff:ff link-netnsid 14
1772: veth99fe34a@if1771: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether 0a:72:7b:26:fe:be brd ff:ff:ff:ff:ff:ff link-netnsid 11
1776: veth1b14864@if1775: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether c6:ae:d2:7a:5d:e9 brd ff:ff:ff:ff:ff:ff link-netnsid 13
1784: vethd8b7cf4@if1783: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether de:f7:74:de:68:e2 brd ff:ff:ff:ff:ff:ff link-netnsid 16
1788: vethc86cd8a@if1787: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker_gwbridge state UP mode DEFAULT group default 
    link/ether e6:17:1d:8d:1c:3f brd ff:ff:ff:ff:ff:ff link-netnsid 17
```

**Comment by ndegory** [comment]
ok, I've seen that in the past, but it was fixed in #1321.
the workaround is to stop the service, remove the volume, and start it again.
what is the context? (OS type, Docker version, cloud provider, how many nodes)

**Comment by freignat91** [comment]
Error due to constraint put on node label where there is no label set for nodes in the cluster.
Close the issue
